### DISCUSSION TITLE
How to specify thread number of each query

### DISCUSSION THREAD

**Discussion by saicomagic** [milvus-io/milvus#15948] [Q&A and General discussion] [answered]
I am using milvus 1.1.1, I want to know how to specify thread number of each query. The CPU consumption is very high.

**Comment by yhmo** [comment, marked as answer]
There is a hidden config item that can specify the parallel thread number of ANNS(Approximate Nearest Neighbor Search).
You can add this item in the server_config.yaml:
```
engine_config:
  omp_thread_num: 8
```
The "omp_thread_num" defines how many threads will be created when the search engine do ANNS.

**Reply by saicomagic** [reply]
Get it! Thank you very much!

**Comment by yushangdi** [comment]
I'm using Milvus 2.3.x. Can I still use this config? Is there another config setting now for specifying thread number of each query or for building the index? Thanks!

**Reply by yhmo** [reply]
Not available in v2.3.x. Not able to config thread number in v2.3.x.
